`-p:NativeLib=Shared` to be implied in the absence of NativeLib

Using `7.0.100-preview.5.22262.14` daily build.

Create a classlib project using `dotnet new classlib` and publish it with the following command:

```sh
dotnet publish --use-current-runtime -p:PublishAot=true
```

it fails with the following error:

```
Microsoft (R) Build Engine version 17.3.0-preview-22256-03+72a076794 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  Restored /home/am11/projects/nativelib/nativelib.csproj (in 195 ms).
/home/am11/.dotnet7/sdk/7.0.100-preview.5.22262.14/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.RuntimeIdentifierInference.targets(216,5): message NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotnet-support-policy [/home/am11/projects/nativelib/nativelib.csproj]
  nativelib -> /home/am11/projects/nativelib/bin/Debug/net7.0/linux-x64/nativelib.dll
  Generating compatible native code. To optimize for size or speed, visit https://aka.ms/OptimizeNativeAOT
EXEC : error : No entrypoint module [/home/am11/projects/nativelib/nativelib.csproj]
  System.Exception: No entrypoint module
     at ILCompiler.Program.Run(String[] args)
     at ILCompiler.Program.Main(String[] args)
/home/am11/.dotnet7/sdk/7.0.100-preview.5.22262.14/Sdks/Microsoft.DotNet.ILCompiler/build/Microsoft.NETCore.Native.targets(270,5): error MSB3073: The command ""/home/am11/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/7.0.0-preview.5.22258.4/tools/ilc" @"obj/Debug/net7.0/linux-x64/native/nativelib.ilc.rsp"" exited with code 1. [/home/am11/projects/nativelib/nativelib.csproj]
```

Fix is to specify `NativeLib` property.

```
dotnet publish --use-current-runtime -p:PublishAot=true -p:NativeLib=Shared
```

It would make sense when NativeLib is unspecified for OutputType != Exe, its value is defaulted to `Shared` by the SDK.

cc @LakshanF, @agocke

--- a/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets
+++ b/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets
@@ -1,6 +1,6 @@
 <!--
 ***********************************************************************************************
-Microsoft.NETCore.Native.Unix.props
+Microsoft.NETCore.Native.Unix.targets
 
 WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and have
           created a backup copy.  Incorrect changes to this file will make it

--- a/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Windows.targets
+++ b/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Windows.targets
@@ -1,6 +1,6 @@
 <!--
 ***********************************************************************************************
-Microsoft.NETCore.Native.Windows.props
+Microsoft.NETCore.Native.Windows.targets
 
 WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and have
           created a backup copy.  Incorrect changes to this file will make it

--- a/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets
+++ b/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets
@@ -16,6 +16,7 @@ The .NET Foundation licenses this file to you under the MIT license.
 
   <!-- Set defaults for unspecified properties -->
   <PropertyGroup>
+    <NativeLib Condition="'$(OutputType)' == 'Library' and '$(NativeLib)' == ''">Shared</NativeLib>
     <NativeIntermediateOutputPath Condition="'$(NativeIntermediateOutputPath)' == ''">$(IntermediateOutputPath)native\</NativeIntermediateOutputPath>
     <NativeOutputPath Condition="'$(NativeOutputPath)' == ''">$(OutputPath)native\</NativeOutputPath>
     <NativeCompilationDuringPublish Condition="'$(NativeCompilationDuringPublish)' == ''">true</NativeCompilationDuringPublish>
@@ -274,8 +275,8 @@ The .NET Foundation licenses this file to you under the MIT license.
     <Touch Condition="$([MSBuild]::VersionLessThan('$(NETCoreSdkVersion)', '7.0.0'))" Files="$(_LinkSemaphore)" AlwaysCreate="true" />
   </Target>
 
-  <Import Project="$(MSBuildThisFileDirectory)Microsoft.NETCore.Native.Windows.props" Condition="'$(TargetOS)' == 'windows'" />
-  <Import Project="$(MSBuildThisFileDirectory)Microsoft.NETCore.Native.Unix.props" Condition="'$(TargetOS)' != 'windows'" />
+  <Import Project="$(MSBuildThisFileDirectory)Microsoft.NETCore.Native.Windows.targets" Condition="'$(TargetOS)' == 'windows'" />
+  <Import Project="$(MSBuildThisFileDirectory)Microsoft.NETCore.Native.Unix.targets" Condition="'$(TargetOS)' != 'windows'" />
 
   <Target Name="MultiFileCopyNative"
       Inputs="@(NativeObjects)"

